### PR TITLE
Add isolation group header to service client

### PR DESCRIPTION
**Pull request by Shaddoll**
**What?**
Add a new header to thrift and grpc client to specify the isolation group of the client.

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/compatibility/proto/serviceclient/GrpcServiceStubs.java [hunk 1]
```diff
@@ -61,6 +62,8 @@ final class GrpcServiceStubs implements IGrpcServiceStubs {
       Metadata.Key.of("cadence-client-feature-version", Metadata.ASCII_STRING_MARSHALLER);
   private static final Metadata.Key<String> CLIENT_IMPL_HEADER_KEY =
       Metadata.Key.of("cadence-client-name", Metadata.ASCII_STRING_MARSHALLER);
+  private static final Metadata.Key<String> ISOLATION_GROUP_HEADER_KEY =
+      Metadata.Key.of("cadence-client-isolation-group", Metadata.ASCII_STRING_MARSHALLER);
```
Comment (davidporter-id-au) [inline src/main/java/com/uber/cadence/internal/compatibility/proto/serviceclient/GrpcServiceStubs.java:66]: I suppose i should make the go one match this

**Review by davidporter-id-au — APPROVED** [review]
Looks good, but can we ensure we do an e2e test before we merge this

File: src/main/java/com/uber/cadence/internal/compatibility/proto/serviceclient/GrpcServiceStubs.java (hunk 1 above)
Comment (Shaddoll) [reply src/main/java/com/uber/cadence/internal/compatibility/proto/serviceclient/GrpcServiceStubs.java:66]: yeah

File: src/main/java/com/uber/cadence/serviceclient/ClientOptions.java [hunk 2]
```diff
@@ -79,6 +79,8 @@ public class ClientOptions {
   private final IAuthorizationProvider authProvider;
   /** Optional Feature flags to turn on/off some Cadence features */
   private final FeatureFlags featureFlags;
+  /** Optional isolaationGroup of the service */
```
Comment (shijiesheng) [inline src/main/java/com/uber/cadence/serviceclient/ClientOptions.java:82]: nit: typo

File: src/main/java/com/uber/cadence/internal/compatibility/proto/serviceclient/GrpcServiceStubs.java (hunk 1 above)
Comment (shijiesheng) [reply src/main/java/com/uber/cadence/internal/compatibility/proto/serviceclient/GrpcServiceStubs.java:66]: in go client, we didn't make the oss change so it needs two changes. (oss + internal fx package)

**Review by shijiesheng — APPROVED** [review]

**Pull request closed without merging**